documentation/design-docs/dotnet-tools.md

```diff
@@ -398,6 +399,45 @@ CONVERT
       Writing:       ./trace.speedscope.json
       Conversion complete
 
+REPORT
+
+    dotnet-trace report [-h|--help]
+                        [-t|--type]
+                        [-p|--parameters]
+                        [-o|--output <output_file_path>]
+                        --format <format>
+                        <trace_file>
+
+    Generate report into stdout from a previously generated trace_file.
+
+    -h, --help
+        Show command line help
+    
+    -t, --type
+        Specifies the type of report to create. The first type of report that will be available is 'topN'
+
+    -p, --parameters
+        Takes a list of arguments for each specific report type, see below for details.
+        type            parameters  explanation
+        'topN'          -n          the number of top functions to gather on the stack
+                        -i          find the top n methods inclusively on the stack. Without this tag, will find the top N methods exclusively on the stack
+        'countEvent'    
+
+    -o, --output
+        The path where the converted file is written. If unspecified, the file is written in the current directory
+        using the same base filename as the input file and the extension appropriate for the new format.
+
+    --format
+        Specifies the format to convert the trace file to.
+
+    <trace_file>
+        The path to the trace file that will be read to generate the report.
+
+    Examples:
+      > dotnet-trace report -t 'topN' -p <-n 10, -i >
+      Writing:       
+      Report complete
```

Comment:
I think the interesting part of this spec would be what the report looks like. In this case, if I request the `TopN` report, what sort of output would I see?

Perhaps
```
Top 5 functions (exclusive):
  1. MyPoorlyPerformantFunction(...) <IL offset etc.>
  2. ...
  3. ...
  4. ...
  5. ...
```

or
```
Top 5 functions (exclusive):
function name                                       Inclusive       Exclusive
MyPoorlyPerformantFunction(...) <IL offset etc.>    20% (100,483)   88% (501,321)
...
...
...
...
```
or
```
Top 5 functions (exclusive):

  MyPoorlyPerformantFunction(...) <IL offset etc.>  Inclusive: N% (####) Exclusive: M% (####)
    - <caller function>                             Inclusive: N% (####) Exclusive: M% (####)
    - <caller function>                             Inclusive: N% (####) Exclusive: M% (####)
    - <caller function>                             Inclusive: N% (####) Exclusive: M% (####)
    - <caller function>                             Inclusive: N% (####) Exclusive: M% (####)
    - 10 more frames

  MyOtherPoorlyPerformantFunction(...)              Inclusive: N% (####) Exclusive: M% (####)
    - <caller function>                             Inclusive: N% (####) Exclusive: M% (####)
    - <caller function>                             Inclusive: N% (####) Exclusive: M% (####)
    - <caller function>                             Inclusive: N% (####) Exclusive: M% (####)
    - <caller function>                             Inclusive: N% (####) Exclusive: M% (####)
    - 5 more frames

  (3 more stacks)
```

Just a few different ways of representing similar data. There isn't a correct way to represent this data, but we should make sure that it is scriptable. Our tools are command line tools and should play nice with the ecosystem of tools like `grep`, `awk`, `sed`, etc. This means we should be consistent with delimiters and what we put on each line of the report.